pkg/syncer/cnsoperator/controller/cnsregistervolume/util.go

```diff
@@ -161,10 +164,43 @@ func getK8sStorageClassNameWithImmediateBindingModeForPolicy(ctx context.Context
 			"Either storagepolicyId: %s doesn't match any storage class, or the policy is not assigned to namespace: %s",
 			storagePolicyID, namespace)
 	} else {
-		// TODO: Implement Quota check using new Storage Quota CR before returning storage class
-		return scName, nil
+		storagePolicyQuotaList := &cnsstoragepolicyquotasv1alpha1.StoragePolicyQuotaList{}
+		err := client.List(ctx, storagePolicyQuotaList, &ctrlruntimeclient.ListOptions{
+			Namespace: namespace,
+		})
+		if err != nil {
+			return "", logger.LogNewErrorf(log, "Failed to list StoragePolicyQuota CR on the namespace: %s", namespace)
+		}
+		log.Debugf("Found scName %s which has matching storagePolicyId %s", scName, storagePolicyID)
+		log.Debugf("Fetch storagePolicyQuotaList: %+v  in namespace %s", storagePolicyQuotaList, namespace)
+		foundMatchStoragePolicyQuotaCR := false
+		matchedStoragePolicyQuotaCR := cnsstoragepolicyquotasv1alpha1.StoragePolicyQuota{}
+		if scName != "" && len(storagePolicyQuotaList.Items) > 0 {
+			for _, storagePolicyQuota := range storagePolicyQuotaList.Items {
+				if storagePolicyQuota.Spec.StoragePolicyId == storagePolicyID {
+					log.Debugf("Found storagePlicyQuota CR with matching storagePolicyId:%s in namespaces:%s",
+						storagePolicyID, namespace)
+					foundMatchStoragePolicyQuotaCR = true
+					matchedStoragePolicyQuotaCR = storagePolicyQuota
+					break
+				}
+			}
+			// NOTE: Below code expects StoragePolicyQuota CR status will be populated with all fields with zero values from the
+			// start (to be taken care by quota controller), otherwise this code will always return error.
```

Comment:
I'm not following this NOTE.  This function is called in a reconcile loop, right? Why isn't it possible for StoragePolicyQuota CR status to be updated already with non-zero values?

Reply:
@xing-yang The check in this code expects StoragePolicyQuota CR status exists since it need to check StoragePolicyQuotaCR.Status.SCLevelQuotaStatuses.StorageClassName to see if it matches "scName" which is found earlier in this function.  Quota controller will update the CR status. It we just create the StoragePolicyQuotaCR manually, it will not have the `.Status` field and the checking in line 190 will not be executed and the function will fail with the following error at line 200.

Reply:
This line "otherwise this code will always return error" is confusing. Which code will always return error and in what case?  Please either clarify or just get rid of this line.